src/main/java/dk/sdu/se_f22/sharedlibrary/models/Product.java

```diff
@@ -1,83 +1,184 @@
+
+        if (!baseProduct.get(ProductAttribute.SIZE).equals("unavailable")){
+            this.size = baseProduct.get(ProductAttribute.SIZE);
         }
-        
-        if (value.isEmpty()){
-            productAttributes.put(pA, null);
-            return productAttributes.get(pA) == null;
+
+        if (!baseProduct.get(ProductAttribute.CLOCKSPEED).equals("unavailable")){
+            this.clockspeed = Double.parseDouble(baseProduct.get(ProductAttribute.CLOCKSPEED));
         }
-        
-        productAttributes.put(pA,value); //assigns key and values to map
-        return productAttributes.get(pA).equalsIgnoreCase(value);
-    }
-    
-    public boolean setLocations(ArrayList<String> values){
-        
-        StringBuilder sB = new StringBuilder();
-        for(String s : values){
-            sB.append(s).append(",");
+
+        if (!baseProduct.get(ProductAttribute.WEIGHT).equals("unavailable")){
+            this.weight = Double.parseDouble(baseProduct.get(ProductAttribute.WEIGHT));
         }
-        productAttributes.put(ProductAttribute.IN_STOCK, sB.toString());
-        
-        return values.size() > 0; //checks if there is stock of the product in any shop
     }
-    
-    @Override
-    public String toString(){ //overrides toString method, returning product name and price
-        return "Product: " + productAttributes.get(ProductAttribute.NAME) + " price: " + productAttributes.get(ProductAttribute.PRICE);
-    }
-    
-    public void print(){
-        System.out.println("Product : " + productAttributes.get(ProductAttribute.NAME)); //returns product name
-        
-        for(ProductAttribute pA : ProductAttribute.values()){ //runs through all product attributes
-            
-            if(pA == ProductAttribute.IN_STOCK){ //if product is in stock, prints the available locations
-                StringBuilder toPrint = new StringBuilder();
-                for(String s : getLocations()){
-                    toPrint.append(s).append("\t");
-                }
-                System.out.println("\t " + pA.alias + ": " + toPrint);
-                
-            }else {
-                
-                System.out.println("\t " + pA.alias + ": " + get(pA));
-            }
+
+    /**@param attributeName The name of the attribute to retrieve
+     * @return The value of the attribute referenced
+     * @throws IllegalArgumentException if the attribute input does not correspond to any of the attributes, whose value is a double
+     */
+    public double getDoubleValue(String attributeName){
+        return switch (attributeName) {
+            case ("price") -> this.getPrice();
+            case ("averageUserReview") -> this.getAverageUserReview();
+            case ("clockspeed") -> this.getClockspeed();
+            case ("weight") -> this.getWeight();
+            default -> throw new IllegalArgumentException(attributeName + "does not exist as a double attribute: ");
+        };
+    }
```

Comment:
Is this method necessary anymore? Since it all has been isolated into individual getters, with proper types.

Reply:
Will remove